File: ShellingLaptopApp/app/src/main/java/com/shellinglaptop/utils/BindingUtils.java
package com.shellinglaptop.utils;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.util.Base64;
import android.view.View;
import android.widget.ImageView;

import androidx.databinding.BindingAdapter;

import com.bumptech.glide.Glide;

public class BindingUtils {
    @BindingAdapter("imageStr")
    public static void loadImage(ImageView imageView, String url){
        Bitmap bitmap = ImageUtils.convertStringToBitmap(url);
        imageView.setImageBitmap(bitmap);
    }

}

File: ShellingLaptopApp/app/src/main/java/com/example/shellinglaptopapp/util/BindingUtils.kt
package com.example.shellinglaptopapp.util

import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import android.widget.ImageView
import android.widget.RelativeLayout
import androidx.databinding.BindingAdapter
import com.bumptech.glide.Glide


@BindingAdapter("imageUrl")
fun loadImage(view: ImageView, url: String){
    Glide.with(view.context)
        .load(url)
        .into(view)
}

@BindingAdapter("app:textWatcher")
fun watcher(view: EditText, textWatcher: TextWatcher){
    view.addTextChangedListener(textWatcher)
}

@BindingAdapter("android:alignParentRight")
fun setLayoutAlignParentRight(view: View, parent: Boolean){
    val layoutParams = view.layoutParams as RelativeLayout.LayoutParams
    layoutParams.addRule(
        RelativeLayout.ALIGN_PARENT_RIGHT,
        if (parent) RelativeLayout.TRUE else 0
    )
    view.layoutParams = layoutParams
}